### PR TITLE
Verify signature of SAML request

### LINKED ISSUE #49: Verify signature on SAML requests
New settings:
* `CONNECTOR_SERVICE_PROVIDER['CERT_FILE]` (optional, default `None`): The path of a certificate to verify the signature of Service Provider's authentication requests.

Implementation:
* `SAMLRequest.request_signature() -> Optional[Element]`: returns `<ds:Signature>` element is there is any.
* `SAMLRequest.verify_request(cert_file: str) -> None`: Verifies the signature. Raises `SecurityError` if there is no signature or the signature is invalid.
* `ServiceProviderRequestView.get_saml_requestcountry_parameter: str, cert_file: Optional[str]) -> SAMLRequest`: Modified to verify the signature if `cert_key` is provided.
* `DemoServiceProviderRequestView.post`: Modified to sign the SAML request reusing `CONNECTOR_SERVICE_PROVIDER['RESPONSE_SIGNATURE']`.
* `SAMLRequest.sign_request(key_file: str, cert_file: str, signature_method: str, digest_method: str) -> None:` Sign SAML request.

### PR DESCRIPTION
**Pull request by fenryxo**
Closes #49

### CODE REVIEW

**Comment by tpazderka** [comment]
This needs a rebase...

**Comment by fenryxo** [comment]
Rebased.

File: eidas_node/saml.py [hunk 1]
```diff
@@ -180,6 +185,31 @@ def create_light_request(self) -> LightRequest:
 
         return request
 
+    def sign_request(self, key_file: str, cert_file: str, signature_method: str, digest_method: str) -> None:
+        """Sign the whole SAML request."""
+        # Remove old signature if there is any
```
Comment (tpazderka) [inline eidas_node/saml.py]: Old signature should cause an error.

Comment (fenryxo) [reply eidas_node/saml.py]: Fixed.

**Pull request merged**